controlplane/src/core/repositories/FeatureFlagRepository.ts

```diff
@@ -580,11 +851,153 @@ export class FeatureFlagRepository {
+      .where(eq(federatedGraphsToFeatureFlagSchemaVersions.baseCompositionSchemaVersionId, baseSchemaVersionId))
+      .execute();
+
+    for (const composition of compositions) {
+      let featureFlagName = '';
+      if (composition.featureFlagId !== null) {
+        const featureFlag = await this.getFeatureFlagById({
+          featureFlagId: composition.featureFlagId,
+          namespaceId,
+        });
+        featureFlagName = featureFlag?.name || '';
+      }
+
+      featureFlagCompositions.push({
+        ...composition,
+        featureFlagName,
+        createdAt: composition.createdAt.toISOString(),
+        compositionErrors: composition.compositionErrors || undefined,
+        createdBy: composition.createdBy || undefined,
+        routerConfigSignature: composition.routerConfigSignature || undefined,
+        admissionError: composition.admissionError || undefined,
+        deploymentError: composition.deploymentError || undefined,
+        isComposable: composition.isComposable || false,
+      });
+    }
+    return featureFlagCompositions;
+  }
+
+  public async getFeatureFlagSchemaVersionsByBaseSchemaVersion({
+    baseSchemaVersionId,
+  }: {
+    baseSchemaVersionId: string;
+  }) {
+    const ffSchemaVersions = await this.db
+      .select({
+        id: federatedGraphsToFeatureFlagSchemaVersions.composedSchemaVersionId,
+        featureFlagId: federatedGraphsToFeatureFlagSchemaVersions.featureFlagId,
+      })
+      .from(federatedGraphsToFeatureFlagSchemaVersions)
+      .where(and(eq(federatedGraphsToFeatureFlagSchemaVersions.baseCompositionSchemaVersionId, baseSchemaVersionId)))
+      .execute();
+
+    if (ffSchemaVersions.length === 0) {
+      return;
+    }
+
+    return ffSchemaVersions;
+  }
+
+  public async getFeatureFlagSchemaVersionByBaseSchemaVersionAndFfId({
+    baseSchemaVersionId,
+    featureFlagId,
+  }: {
+    baseSchemaVersionId: string;
+    featureFlagId: string;
+  }) {
+    const schemaVersion = await this.db
+      .select({
+        id: federatedGraphsToFeatureFlagSchemaVersions.composedSchemaVersionId,
+      })
+      .from(federatedGraphsToFeatureFlagSchemaVersions)
+      .where(
+        and(
+          eq(federatedGraphsToFeatureFlagSchemaVersions.baseCompositionSchemaVersionId, baseSchemaVersionId),
+          eq(federatedGraphsToFeatureFlagSchemaVersions.featureFlagId, featureFlagId),
+        ),
+      )
+      .execute();
+
+    if (schemaVersion.length === 0) {
+      return;
+    }
+
+    const federatedGraphRepo = new FederatedGraphRepository(this.logger, this.db, this.organizationId);
+    const ffSchemaVersion = await federatedGraphRepo.getSchemaVersionById({ schemaVersionId: schemaVersion[0].id });
+
+    return ffSchemaVersion;
+  }
+
+  public async getFeatureSubgraphsByFeatureFlag({
+    featureFlagId,
+    namespaceId,
+  }: {
+    namespaceId: string;
+    featureFlagId: string;
+  }) {
+    const featureSubgraphsByFf = await this.db
```

Comment:
Do we need to filter for the orgId here?

Reply:
Done